src/pages/career/[opportunityId].tsx

```diff
@@ -1,19 +1,35 @@
-import { useRouter } from 'next/router';
+import { NextPage, NextPageContext } from 'next';
 import React from 'react';
 
 import DetailView from 'career/containers/DetailView';
-import CareerOpportunities from 'career/providers/CareerProvider';
 
-const CareerDetailPage = () => {
-  const router = useRouter();
-  const opportunityId = Number(router.query.opportunityId);
+import { careerOpportunitySelectors, fetchCareerOpportunityById } from 'career/slices/careerOpportunities';
+import { Store } from 'core/redux/Store';
+
+interface IContext extends NextPageContext {
+  store: Store;
+}
```

Comment:
Will all pages with getInitialProps have this as interface?
If yes, could this be generalized somewhere and exported?

Reply:
Yes, all pages will have it.
I agree, should be generalized, just didn't think to do it yet :P